Xamarin.Forms.Core/HandlerAttribute.cs

```diff
@@ -5,19 +5,60 @@ namespace Xamarin.Forms
 	[AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
 	public abstract class HandlerAttribute : Attribute
 	{
+		protected HandlerAttribute(Type handler, Type target, object[] filteredSdkVersionInformation) : this(handler, target)
+		{
+			FilteredSdkVersionInformation = filteredSdkVersionInformation;
+			Validate();
+		}
+
 		protected HandlerAttribute(Type handler, Type target)
 		{
 			TargetType = target;
 			HandlerType = handler;
 		}
 
-		internal Type HandlerType { get; private set; }
+		internal Type HandlerType { get; }
+
+		internal Type TargetType { get; }
 
-		internal Type TargetType { get; private set; }
+		internal object[] FilteredSdkVersionInformation { get; }
+
+		protected internal int? MajorVersion { get; protected set; }
 
 		public virtual bool ShouldRegister()
 		{
+			if (FilteredSdkVersionInformation == null || MajorVersion == null)
+				return true;
+
+			var isSupported = (bool)FilteredSdkVersionInformation[0];
+			var minimumSdkVersion = (int)FilteredSdkVersionInformation[1];
+			var maximumSdkVersion = (int)FilteredSdkVersionInformation[2];
+
+			if(minimumSdkVersion <= MajorVersion.Value && MajorVersion.Value <= maximumSdkVersion)
+				return isSupported;
+
+			// This means either the OS version has not originally fallen into the supplied range or 
+			// the OS version has changed and the supplied range is no longer valid.
```

Comment:
Not sure if we should throw an exception here or return true so that the renderer is registered.